test/integral_tests.jl

```diff
@@ -83,7 +83,7 @@ end
 
 @testset "LagrangeInterpolation" begin
     u = [1.0, 4.0, 9.0]
-    t = [1.0, 2.0, 3.0]
+    t = [1.0, 2.0, 6.0]
```

Comment:
Why was this changed?

Reply:
I added this function to conveniently test whether the implementation of a new interpolation type is complete: https://github.com/SciML/DataInterpolations.jl/pull/274/files#diff-f66955aae2aee262495ff068999fe35f8501cd1cbef7b1057aa26d2f1099a4f3R6-R17

To do this, I moved throwing an error when integration is not defined from specialised methods of `integral` to methods of `_integral`. However, that now means that the extrapolation error is thrown before the 'integral not found' error. So, to still test the integral not found error, I had to make sure that the test for this does not try to extrapolate.